tests/test_api.py

```diff
@@ -354,3 +356,127 @@ def test_if_describe_on_remote_git_repo_then_return_expected_info():
         "path": "models/churn.pkl",
         "virtual": False,
     }
+
+
+def test_if_register_with_auto_push_then_invoke_git_push_tag(repo_with_artifact):
+    repo, _ = repo_with_artifact
+    with patch("gto.registry.git_push_tag") as mocked_git_push_tags:
+        gto.api.register(
+            repo=repo.working_dir, name="model", ref="HEAD", auto_push=True
+        )
+    mocked_git_push_tags.assert_called_once_with(
+        repo_path=Path(repo.working_dir).as_posix(),
+        tag_name="model@v0.0.1",
+        delete=False,
+    )
+
+
+def test_if_assign_with_auto_push_then_invoke_git_push_tag(repo_with_artifact):
+    repo, _ = repo_with_artifact
+    with patch("gto.registry.git_push_tag") as mocked_git_push_tags:
+        gto.api.assign(
+            repo.working_dir, name="model", stage="dev", ref="HEAD", auto_push=True
+        )
+    mocked_git_push_tags.assert_called_once_with(
+        repo_path=Path(repo.working_dir).as_posix(),
+        tag_name="model#dev#1",
+        delete=False,
+    )
+
+
+def test_if_unassign_with_auto_push_then_invoke_git_push_tag(repo_with_artifact):
+    repo, _ = repo_with_artifact
+    gto.api.assign(
+        repo.working_dir, name="model", stage="dev", ref="HEAD", auto_push=False
+    )
+    with patch("gto.registry.git_push_tag") as mocked_git_push_tags:
+        # TODO: ask alex why this does not work without specifying ref="HEAD"
```

Comment:
Before merging: could you please help me here?

Reply:
forgot to cite @aguschin

Reply:
Next lines are
```python
        gto.api.unassign(
            repo.working_dir, name="model", stage="dev", ref="HEAD", auto_push=True
        )
```
You unassign a `stage` from a `model version`. Generally, you need to specify the version OR `ref`, that pins down the right version. With `Envs-like` approach we use now, you could remove this limitation, since there is only one model version in a stage at particular moment. But, GTO also tries to support `Kanban-like` approach and `Label-like` approach - [docs](https://github.com/iterative/gto#enabling-multiple-versions-in-the-same-stage-workflow). In those cases you need to keep this limitation in place. That's why it's here. Also, for the reference, we have an [issue somehow related to this](https://github.com/iterative/gto/issues/272).

Reply:
thanks for the clarification, I removed the comment from the code